src/main/java/com/cst438/controller/AssignmentController.java

```diff
@@ -75,21 +133,54 @@ public List<GradeDTO> getAssignmentGrades(@PathVariable("assignmentId") int assi
 
         // TODO remove the following line when done
 
+        Assignment assignment = assignmentRepository.findById(assignmentId).orElse(null);
+        List<Enrollment> enrollments = enrollmentRepository.findEnrollmentsBySectionNoOrderByStudentName(assignment.getSection().getSectionNo());
+        List<GradeDTO> gradeDTOS = new ArrayList<>();
+        for (Enrollment e : enrollments) {
+            Grade grade = gradeRepository.findByEnrollmentIdAndAssignmentId(e.getEnrollmentId(), assignmentId);
+            if (grade == null) {
+                grade.setScore(null);
```

Comment:
NullPointerException

Reply:
Can you go into more detail please? What would cause a NullPointerException here?

Reply:
Fixed with commit [8ee9dd7](https://github.com/jdolarcsumb/CST438-Assignment2-02/commit/8ee9dd7028c2f4e36b7f0912a2fbf4ed8110a42b)